### PR TITLE
Fix launch protocol

### PR DESCRIPTION
**Pull request by dragonDScript**
----------------------------------

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Created / Updated Tests
- [ ] Created / Updated documentation

Tested on a Clean Install with AppImage:
- [ ] Login
- [ ] Download a Game
- [ ] Import a Game
- [ ] Launch Default config
- [ ] Launch Personalized Config
- [ ] Launch wine
- [ ] Launch Proton
- [ ] Launch Custom Proton
- [ ] Launch Tools (Winetricks and Winecfg)
- [ ] Uninstall Game
- [ ] Move Game
- [ ] Languages (Test if no translation stopped working)

### CODE REVIEW

File: electron/protocol.ts [hunk 1]
```diff
@@ -6,24 +6,18 @@ export async function handleProtocol(window : BrowserWindow, url : string) {
...
+    return console.log('Received ping! Arg:', arg)
   }
-  const appName = args.get('appName')
-  const game = Game.get(appName)
   if (command === 'launch') {
+    const game = Game.get(arg)
     const { is_installed } = await game.getGameInfo()
     if (!is_installed) {
-      console.log(`ProtocolHandler: "${appName}" not installed, ignoring launch request.`)
+      console.log(`ProtocolHandler: "${arg}" not installed, ignoring launch request.`)
```
Comment (flavioislima) [inline electron/protocol.ts:18]: We could open a new dialog here saying that the game is not installed and ask to install. 
then we could send the install command to the frontend with the path argument 'default'.

File: electron/protocol.ts [hunk 2]
```diff
@@ -1,29 +1,51 @@
...
+      console.log(`ProtocolHandler: "${arg}" not installed.`)
+      const diag = await dialog.showMessageBox(window, {
+        buttons: [i18next.t('box.yes'), i18next.t('box.no')],
+        cancelId: 1,
+        icon: nativeImage.createFromDataURL(art_logo),
+        message: `${title} ${i18next.t('box.protocol.install.not_installed')}`,
+        title: title
+      })
+      if (diag.response === 0) {
+        window.webContents.send('install', {
```
Comment (flavioislima) [inline electron/protocol.ts:32]: No I think might be confusing what I said, but here you can send the install command only with the appname, and on the frontend, on the GlobalState.tsx file, that line I said above, you need to copy paste how we are handling the launch, but instead call the install command with these arguments, but no on the backend, on the frontend.
Its two different things.
or maybe just call install, appName here and I can handle the frontend later.

Comment (flavioislima) [reply electron/protocol.ts:32]: Yeah, change this line to `...send('install', app_name), and I'll handle the rest.

**Review by flavioislima — APPROVED** [review]
Good job!

**Review by adityaruplaha — CHANGES_REQUESTED** [review]
Please retain the map implementation, passing multiple arguments will be difficult in the future otherwise.

**Comment by dragonDScript** [comment]
> Please retain the map implementation, passing multiple arguments will be difficult in the future otherwise. 

Ok

**Comment by flavioislima** [comment]
> Please retain the map implementation, passing multiple arguments will be difficult in the future otherwise.

No, I have to disagree with that. I don't see any use case for that map function.
99.9% of the time, the protocol will be used for launching game, once you  do your PR, you can adapt to the other needs you think we need, but for now, the current solution @dragonDScript did is perfect.

**Comment by flavioislima** [comment]
It is also made to work with the #443 that he did.
So this should be enough.

**Comment by dragonDScript** [comment]
In the future, though, if there is any use for arguments i Will add It myself.

**Pull request merged**